lib/components/Forms/Fields/NumberInput/index.stories.tsx

```diff
@@ -0,0 +1,58 @@
+import type { Meta, StoryObj } from "@storybook/react"
+
+import { useState } from "react"
+import { NumberInput } from "."
+
+const meta = {
+  render: (props) => <NumberInput key={JSON.stringify(props)} {...props} />,
+  tags: ["autodocs"],
+  args: {
+    disabled: false,
+    placeholder: "Placeholder text here",
+    locale: "en-US",
+  },
+  argTypes: {
+    value: {
+      control: { type: "number" },
+    },
+    locale: {
+      options: ["en-US", "es-ES", "pt-BR"],
```

Comment:
Wondewring if we can type the allowed locales with a TypeScript `enum`?

Reply:
TS doesn't offer an `enum` or `union` with the locales supported by `Intl`, so I think the only way would be to define a union [like this](https://gist.github.com/g4rcez/e072eb9e1203679645985a9c89722117#file-locale-ts) (or with just the main locales used in Factorial). 

However, something to have in mind is that `factorial-i18n` doesn't type the locale as well, so when rendering this component inside Factorial we'd need to assert the type: `<NumberInput locale={i18n.locale as Locales} />`

WDYT?

Reply:
Okay, I understand. If a locale list was somehow standardized, that would probably be the way to go - but since it's not the case, better to leave it as a string and provide some options just on storybook, exactly like you did.

The other way around looks safer at a first glance, but it's equally dangerous for the reasons you mention (bound to do an assertion at one point or another!).